File: Sean的leetcode/Leetcode/Plus One.cpp
d//Time complexity:O(n)
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n = digits.size();

        for (int i = n - 1; i >= 0; i--)
        {
            if (digits[i] == 9)
                digits[i] = 0;
            else
            {
                //if no carry happens, we can return the result immediately.
                digits[i]++;
                return digits;
            }
        }
        //if there's no early return, which means carry still exists.
        digits[0] = 1;
        digits.push_back(0);
        return digits;
    }
};

File: MyLeetCode/python/Plus One.py
###
# Math
# Time Complexity: Worst case: O(n) 
# Space Complexity: O(1)
###
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        if not digits:
            return []
        i = len(digits) - 1
        carry = 1
        while i >= 0:
            num = digits[i] + carry
            carry = num / 10
            digits[i] = num % 10
            if not carry:
                return digits
            i -= 1
        if carry:
            digits = [carry] + digits
        return digits
            
                                        
                            
                    